### ISSUE TITLE
--input-file or similar

### ISSUE THREAD

**Issue by JuhaJGamer** [alexhallam/tv#49]
Any program reading files as input should support deciding where that input comes from. Currently, `tv` seems to function by always opening and reading from stdin. This is inefficient and unconventional, and promoted UUOC. 

Compare the current method:
```
$ cat file | tv

                v write 1         v write 2
file --> (cat) --> pipe --> (tv) --> stdout
      ^ read 1           ^ read 2

Reads:  2
Writes: 2
```

to a better method:

```
$ tv -i file

               v write 1
file --> (tv) --> stdout
      ^ read 1

Reads:  1
Writes: 1
```

It's less memory intensive, it's quicker on older machines, and it follows convention. 

Alternatively, you can still do this on the shell without abusing a concatenation program as a file-reading one. The shell itself is capable of redirecting stdin to be the file itself. It would be at least better for the examples to show the "proper", or at least, better way of reading a file into a program:

```
$ tv < file

              v write 1
file --> (tv) --> stdout
     ^ read 1

Reads:  1
Writes: 1
```

Instead of giving you a pipe or the console input buffer, the shell now gives you the file's file descriptor when you ask for stdin. Readability and the clean "flow" of information through a pipeline isn't an issue here as `tv` is definitely not meant to be piped into other programs, though you could if you wanted to.

TL;DR: Reading from stdin promotes UUOC which makes using the program more intensive, especially noticeably for large files on older machines. Allowing the user to choose the input file, or at least promoting the usage of redirects over `cat` can speed up reading by however long it takes `cat` to read and write your file, and it's less memory intensive. 

Stop `cat` abuse, and stop promoting `cat` abuse!

**Comment by JuhaJGamer** [comment]
> I have never heard of "UUOC" what is that?

Useless use of cat, it's when you use cat when you really don't need to.
There are definitely situations where you'd want to use cat to spool a file in though, it's a lot neater to write `cat file | p1 | p2 | p3 > out` than `p1 < file | p2 | p3 > out` since in one you have a very clear left-to-right flow of data. It's just not necessary here.

**Comment by UlazkaMateusz** [comment]
I wonder if there should be flag like JuhaJGamer suggested. I think passing filename like in cat would be more intuitive.
`$ tv file`
Insted of
`$ tv -i file`
